tools/Az.Tools.Predictor/Az.Tools.Predictor/Commands/BasePSCmdlet.cs

```diff
@@ -0,0 +1,68 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.ApplicationInsights;
+using Microsoft.Azure.PowerShell.Tools.AzPredictor.Utilities;
+using System.Globalization;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.PowerShell.Tools.AzPredictor
+{
+    /// <summary>
+    /// Provides the common base class for creating a ps cmdlet.
+    /// </summary>
+    public abstract class BasePSCmdlet : PSCmdlet
+    {
+        /// <inheritdoc/>
+        protected override void BeginProcessing()
+        {
+            TelemetryClient telemetryClient = TelemetryUtilities.CreateApplicationInsightTelemetryClient();
+            var settings = Settings.GetSettings();
+            var azContext = new AzContext()
+            {
+                IsInternal = (settings.SetAsInternal == true) ? true : false,
```

Comment:
Just set it to `settings.SetAsInterna`? Is it done this way, because it could be `null` or something?

Reply:
`SetAsInternal` can be null. It's an optional settings. This means `IsInternal` should only be set to true when `SetAsInternal` is set to true.